### ISSUE TITLE
BUG : NPC threat on pull

### ISSUE THREAD

**Issue by Maurowin** [azerothcore/azerothcore-wotlk#5874]
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [ ] enUS
- [X] other (frFR)

##### FACTION

- [X] Alliance
- [ ] Horde

##### CONTENT PHASE:

- [x] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

##### CURRENT BEHAVIOUR:

When I miss my pull attack, I don't take the NPC's aggro
![WoWScrnShot_051421_200220](https://user-images.githubusercontent.com/31663279/118311502-481ddf00-b4f0-11eb-9941-6786378bf1bf.jpg)

##### EXPECTED BLIZZLIKE BEHAVIOUR:

When I miss my pull attack, I have to take the aggro of the NPC

##### STEPS TO REPRODUCE THE PROBLEM:

1. Miss attack on the pull like Judgment 

##### EXTRA NOTES:

Type here.

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/8dd58ebb5bea6a8aada54c830d06974d3ecd1353

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

**Comment by UltraNix** [comment]
@Maurowin I can't reproduce it. Tell me what spell I should use.

**Comment by Maurowin** [comment]
U can use Judgment 
https://www.wowhead.com/spell=20271/judgment

But you must miss your attack.

**Comment by UltraNix** [comment]
I don' know how to miss with spell - I see only resists.

**Comment by Maurowin** [comment]
u should try several time until miss :)
And when the npc resist he attack you ?

**Comment by UltraNix** [comment]
Yes - he attacks me.

**Comment by UltraNix** [comment]
Can't confirm it.
![asdf](https://user-images.githubusercontent.com/80540499/126676316-dde11261-696d-4404-a191-e3120e8d91f5.png)

**Comment by FrancescoBorzi** [comment]
@azerothcore/bug-triaging can anyone reproduce the issue? otherwise we should close it

**Comment by syssneck** [comment]
Hi, let me try it later or latest tomorrow!

**Comment by Maurowin** [comment]
> 
> 
> Can't confirm it.
> ![asdf](https://user-images.githubusercontent.com/80540499/126676316-dde11261-696d-4404-a191-e3120e8d91f5.png)

I seen on your battle log, u reach to attack him first before miss spell.

**Comment by UltraNix** [comment]
No? Only Seal is refreshed and then casted Judgement that has been missed.

**Comment by Maurowin** [comment]
I'm uploading video on youtube from ChromieCraft at current version

**Comment by Maurowin** [comment]
look here : https://youtu.be/m0M7vsnjwbA

**Comment by UltraNix** [comment]
What is this mob?

**Comment by benjymansy123** [comment]
> What is this mob?

The mob in his video is [Boulderheart](https://classic.wowhead.com/npc=14273/boulderheart).

**Comment by UltraNix** [comment]
And what is your level?

**Comment by UltraNix** [comment]
![fasfqw](https://user-images.githubusercontent.com/80540499/126696822-72c2773c-1c01-4c0f-982a-4ed72c093d44.png)

**Comment by syssneck** [comment]
So, I tested it multiple times, and he indeed completely ignores me, when I miss him with one of the Judgements. I also tried with autoattacks, but he always attacked me, even when I missed with an autoattack. I'm on build (d187ba1). The Level was 17-20.

https://user-images.githubusercontent.com/68868567/126698781-20984127-b55f-424f-901a-dc620367def9.mp4

**Comment by Maurowin** [comment]
Iam lvl 20

**Comment by UltraNix** [comment]
@syssneck Indeed - the miss puts you in combat but this mob completely ignored you.

**Comment by syssneck** [comment]
@UltraNix but I reproduced it on my local AC, this is strange. Is your server up to date?

**Comment by UltraNix** [comment]
@syssneck Yes, it is.

**Comment by Maurowin** [comment]
Is not only with this NPC. All neutrals are like that. Can't test with aggrives NPC because i body pull all time.

**Comment by UltraNix** [comment]
Okay.
All judgement spells have this attr: `SPELL_ATTR3_NO_INITIAL_AGGRO`. It means that casting these spells cause not aggro nor threat.
So if the spell is missed, so the mob (neutral or hostile) will not aggro on us.
If the spell hits, so the triggered oe (the real one that is applied) causes to aggro on us (it does not have that attr.)

TLDR: It works okay. Now it's time to find the proof that mobs should aggro if the judgement spell missed.

**Comment by UltraNix** [comment]
I will close this soon if no proof wil be found.

**Comment by Maurowin** [comment]
> 
> 
> Okay.
> All judgement spells have this attr: `SPELL_ATTR3_NO_INITIAL_AGGRO`. It means that casting these spells cause not aggro nor threat.
> So if the spell is missed, so the mob (neutral or hostile) will not aggro on us.
> If the spell hits, so the triggered one (the real one that is applied) causes to aggro on us (it does not have that attr.)
> 
> TLDR: It works okay. Now it's time to find the proof that mobs should aggro if the judgement spell missed.

That's the problem, this spell should not be with this parameter.

On classic, even if we miss, we take the aggro.

**Comment by T1ti** [comment]
SPELL_ATTR3_NO_INITIAL_AGGRO was incorrect description and deosn't exist anymore.
The TBC classic client leaked the spell attribute definitions and it was updated to "SPELL_ATTR3_SUPRESS_TARGET_PROCS" 
![image](https://user-images.githubusercontent.com/40864460/131389299-e205d6e4-0824-4626-b91e-428b72725f22.png)

**Comment by T1ti** [comment]
So this is just compeltely wrong implementation of that flag.
862 spells currently use that flag and ahve no threat because of it, I noticed a lot of class spells that shouldn't ahve it like Warrior's Slam, paladin's Holy shock etc...
Those spells probably do no 0 aggro atm
![image](https://user-images.githubusercontent.com/40864460/131391623-13971e92-d384-4610-969a-13fd1c01b77b.png)
![image](https://user-images.githubusercontent.com/40864460/131391644-d6eea095-77be-4480-a469-13ba42ace324.png)

**Comment by UltraNix** [comment]
So @T1ti is retaking/continuing this in #7649

**Comment by T1ti** [comment]
I don't think my PR fixes this at all, it's just that I don't think the flag is invalidating this issue
Also I don't think that's only an issue with spelsl that use the flag ?
